pkg/sql/schemachanger/corpus/corpus_test: TestValidateCorpuses failed

pkg/sql/schemachanger/corpus/corpus_test.TestValidateCorpuses [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/14772149?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/14772149?buildTab=artifacts#/) on master @ [911f1ce389342459592d89fe7b78bcba3a2d265a](https://github.com/cockroachdb/cockroach/commits/911f1ce389342459592d89fe7b78bcba3a2d265a):


```
=== RUN   TestValidateCorpuses
    corpus_test.go:41: 
        	Error Trace:	pkg/sql/schemachanger/corpus/corpus_test_test/pkg/sql/schemachanger/corpus/corpus_test.go:41
        	Error:      	Received unexpected error:
        	            	open /artifacts/corpus-mixed/corpus: no such file or directory
        	Test:       	TestValidateCorpuses
--- FAIL: TestValidateCorpuses (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestValidateCorpuses.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37690

--- a/build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh
+++ b/build/teamcity/cockroach/nightlies/sqllogic_corpus_nightly_impl.sh
@@ -75,7 +75,7 @@ if [ $exit_status = 0 ]; then
 fi
 
 # Generate a corpus for all mixed version variants
-for config in local-mixed-23.1; do
+for config in local-mixed-23.2; do
   $BAZEL_BIN/pkg/cmd/bazci/bazci_/bazci test -- --config=ci \
       //pkg/sql/logictest/tests/$config/... \
       --test_arg=--declarative-corpus=$ARTIFACTS_DIR/corpus-mixed \

